src/AppInstallerCommonCore/Public/winget/ManifestCommon.h

```diff
@@ -112,20 +112,58 @@ namespace AppInstaller::Manifest
         Preview,
     };
 
-    struct PackageDependency
+    enum class DependencyType
     {
-        string_t Id;
-        string_t MinVersion;
+        WindowsFeature,
+        WindowsLibraries,
+        Package,
+        External
     };
 
     struct Dependency
     {
-        std::vector<string_t> WindowsFeatures;
-        std::vector<string_t> WindowsLibraries;
-        std::vector<PackageDependency> PackageDependencies;
-        std::vector<string_t> ExternalDependencies;
+        DependencyType Type;
+        string_t Id;
+        std::optional<string_t> MinVersion;
+
+        Dependency(DependencyType type, string_t id, std::optional<string_t> minVersion) : Type(type), Id(id), MinVersion(minVersion) {}

```

Comment:
The parameter here should be `string_t minVersion` as there is no need to call this overload if you don't have a value for it.

All of these should use `std::move` to initialize the string value from the parameter.